packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -2426,7 +2426,7 @@ export class EntityService {
             relatedEntityId,
             relatedFieldId,
             // eslint-disable-next-line @typescript-eslint/naming-convention
-            fkHolder,
+            fkHolder: relatedFieldId || fkHolder,
```

Comment:
Amit can you please explain this change. I have concern about two things:
1. are we using the correct value when we use relatedFieldId? I remember that fkholder should use the permanentId 
2. in which use case, we are using the relatedFieldId and not the value of fkHolder?

Reply:
I can show you in the code that although we call it like that, we use it in another place as the permanent id

Reply:
![image](https://github.com/amplication/amplication/assets/39680385/fd2dc306-4456-4b05-a0bf-dbf637c3b860)

![image](https://github.com/amplication/amplication/assets/39680385/652788cd-fee3-4669-b66f-b5a84a1a9087)

Reply:
thanks @abrl91 
I didnt see a reply for the second concern. Since there is already a parameter that is called fkHolder - I would expect the calling function to send the value to that parameter instead of doing this local change inside the function. 
Please think wether this change is really required. If you are sure this is required and ok with it you can resolve this comment

Reply:
but since we don't create the permanentId in the schema utils, I can't its value. The value is set only on the entity service in the create related field function